ui/components/MesheryPlayComponent.js

```diff
@@ -83,217 +94,195 @@ const styles = (theme) => ({
   },
 });
 
-class MesheryPlayComponent extends React.Component {
-  constructor(props) {
-    super(props);
-
-    const { meshAdapters } = props;
-    let adapter = {};
-    if (meshAdapters && meshAdapters.size > 0) {
-      adapter = meshAdapters[0];
-    }
-    this.state = {
-      adapter,
-    };
-  }
+const MesheryPlayComponent = ({ meshAdapters, setAdapter, classes }) => {
+  const [adapter, setAdapterState] = useState(null);
+  const router = useRouter();
+  const prevMeshAdaptersRef = useRef();
 
-  handleRouteChange = () => {
-    const queryParam = this.props?.router?.query?.adapter;
+  const handleRouteChange = () => {
+    const queryParam = router?.query?.adapter;
     if (queryParam) {
-      const selectedAdapter = this.props.meshAdapters.find(
+      const selectedAdapter = meshAdapters.find(
         ({ adapter_location }) => adapter_location === queryParam,
       );
       if (selectedAdapter) {
-        this.setState({ adapter: selectedAdapter });
+        setAdapterState(selectedAdapter);
       }
-    } else if (this.props.meshAdapters.size > 0) {
-      this.setState({ adapter: this.props.meshAdapters.get(0) });
+    } else if (meshAdapters.size > 0) {
+      setAdapterState(meshAdapters.get(0));
     }
   };
 
-  componentDidMount() {
-    const { router } = this.props;
-    router.events.on('routeChangeComplete', this.handleRouteChange);
-  }
+  useEffect(() => {
+    // componentDidMount
```

Comment:
nit: we can probably remove these mapping comments.

Reply:
Done